From c35f9f15189c4a0621818db32f42de7984430b96 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Thu, 27 Mar 2025 19:35:49 +0100
Subject: [PATCH 1646/1695] plat-stm32mp1: add stm32_exti_base function

Add a function to access to EXTI base address.

This function is introduced only for STM32MP1 family, with only
one EXTI instance and it is a preliminary to cleanup the EXTI driver.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 3e94e49ad7806daea5ef4a399fb4350ebd7bc0ad plat-stm32mp1: add stm32_exti_base function]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I34926a19e28982f4aee5c6aeb29e9d0f56ebe463
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/447277
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/plat-stm32mp1/main.c            | 7 +++++++
 core/arch/arm/plat-stm32mp1/platform_config.h | 1 +
 core/include/drivers/stm32mp1_rcc_util.h      | 2 ++
 3 files changed, 10 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp1/main.c b/core/arch/arm/plat-stm32mp1/main.c
index eef2bedb5..fa919d4b9 100644
--- a/core/arch/arm/plat-stm32mp1/main.c
+++ b/core/arch/arm/plat-stm32mp1/main.c
@@ -451,6 +451,13 @@ vaddr_t stm32_rcc_base(void)
 	return io_pa_or_va_secure(&base, 1);
 }
 
+vaddr_t stm32_exti_base(void)
+{
+	static struct io_pa_va base = { .pa = EXTI_BASE };
+
+	return io_pa_or_va_secure(&base, 1);
+}
+
 vaddr_t get_gicd_base(void)
 {
 	struct io_pa_va base = { .pa = GIC_BASE + GICD_OFFSET };
diff --git a/core/arch/arm/plat-stm32mp1/platform_config.h b/core/arch/arm/plat-stm32mp1/platform_config.h
index 257ba85c4..5af5cbfe8 100644
--- a/core/arch/arm/plat-stm32mp1/platform_config.h
+++ b/core/arch/arm/plat-stm32mp1/platform_config.h
@@ -65,6 +65,7 @@
 #define ETH1_BASE			0x5800A000
 #define ETH2_BASE			0x5800E000
 #define ETZPC_BASE			0x5c007000
+#define EXTI_BASE			0x5000d000
 #define FMC_BASE			0x58002000
 #define GIC_BASE			0xa0021000ul
 #define GPIOA_BASE			0x50002000
diff --git a/core/include/drivers/stm32mp1_rcc_util.h b/core/include/drivers/stm32mp1_rcc_util.h
index e26345042..ff7138857 100644
--- a/core/include/drivers/stm32mp1_rcc_util.h
+++ b/core/include/drivers/stm32mp1_rcc_util.h
@@ -10,6 +10,8 @@
 vaddr_t stm32_rcc_base(void);
 void stm32_reset_system(void);
 
+vaddr_t stm32_exti_base(void);
+
 /* Helper from platform RCC clock driver */
 struct clk *stm32mp_rcc_clock_id_to_clk(unsigned long clock_id);
 unsigned int stm32mp_rcc_clk_to_clock_id(struct clk *clk);
-- 
2.47.2

